project-squirrel-dragon/PresentationLayer/Modules/CardDetailsModule/CardDetailViewModel.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  CardsDetailViewModel.swift
+//  project-squirrel-dragon
+//
+//  Created by Justine Wright on 2021/11/15.
+//
+
+import Foundation
+
+class CardDetailViewModel {
+    private var cardsCollected: uint = 0
+    private (set) var cardId: String = ""
+    private (set) var collectableCard: CollectableCard!
+    private var originalCollected: uint!
+
+    public func configure(withCard card: CollectableCard) {
+        collectableCard = card
+        totalCardsCollected = card.userCard?.collectedNumber ?? 0
+        cardId = card.id
+        originalCollected = cardsCollected
+    }
+
+    public func addOneCard() {
+        totalCardsCollected += 1
+    }
+
+    public func removeOneCard() {
+        totalCardsCollected -= 1
+    }
+
+    var finalCardsCollected: Int? {
+        return cardsCollected == originalCollected ? nil : totalCardsCollected
+    }
+
+    var totalCardsCollected: Int {
+        get {
+            return Int(cardsCollected)
+        }
+
+        set {
+            cardsCollected = newValue < 0 ? 0 : uint(newValue)
+        }
+    }
+
+    var nationalPokedexNumberLabel: String? {
+        guard let number = collectableCard.pokemonCard.nationalPokedexNumbers.first else {
```

Comment:
Just a code style thing, but might look a little better to put the `return nil` on the same line.